src/Microsoft.ML.Onnx/OnnxContextImpl.cs

```diff
@@ -0,0 +1,257 @@
+
+            _nodes = new List<NodeProto>();
+            _intermediateValues = new List<OnnxUtils.ModelArgs>();
+            _inputs = new List<OnnxUtils.ModelArgs>();
+            _outputs = new List<OnnxUtils.ModelArgs>();
+            _columnNameMap = new Dictionary<string, string>();
+            _variableNames = new HashSet<string>();
+            _nodeNames = new HashSet<string>();
+            _name = name;
+            _producerName = producerName;
+            _producerVersion = producerVersion;
+            _modelVersion = modelVersion;
+            _domain = domain;
+        }
+
+        public override bool ContainsColumn(string colName) => _columnNameMap.ContainsKey(colName);
+
+        /// <summary>
+        /// Stops tracking a column. If removeVariable is true then it also removes the 
+        /// variable associated with it, this is useful in the event where an output variable is 
+        /// created before realizing the transform cannot actually save as ONNX.
+        /// </summary>
+        /// <param name="colName">IDataView column name to stop tracking</param>
+        /// <param name="removeVariable">Remove associated ONNX variable at the time.</param>
+        public override void RemoveColumn(string colName, bool removeVariable)
+        {
+            _host.CheckNonEmpty(colName, nameof(colName));
+
+            if (removeVariable)
+            {
+                foreach (var val in _intermediateValues)
+                {
+                    if (val.Name == _columnNameMap[colName])
+                    {
+                        _intermediateValues.Remove(val);
+                        break;
+                    }
+                }
+            }
+
+            if (_columnNameMap.ContainsKey(colName))
+                _columnNameMap.Remove(colName);
+        }
+
+        /// <summary>
+        /// Removes an ONNX variable. If removeColumn is true then it also removes the 
+        /// IDataView column associated with it.
+        /// </summary>
+        /// <param name="variableName">ONNX variable to remove.</param>
+        /// <param name="removeColumn">IDataView column to stop tracking</param>
+        public override void RemoveVariable(string variableName, bool removeColumn)
+        {
+            _host.CheckNonEmpty(variableName, nameof(variableName));
+            if (!_columnNameMap.ContainsValue(variableName))
+                throw _host.ExceptParam(nameof(variableName), $"Could not find '{variableName}' declared in ONNX graph");
+
+            if (removeColumn)
+            {
+                foreach (var val in _intermediateValues)
+                {
+                    if (val.Name == variableName)
+                    {
+                        _intermediateValues.Remove(val);
+                        break;
+                    }
+                }
+            }
+
+            string columnName = _columnNameMap.Single(kvp => string.Compare(kvp.Value, variableName) == 0).Key;
```

Comment:
>string.Compare(kvp.Value, variableName) == 0 [](start = 61, length = 44)

curious, why you prefer string.Compare to just '=='

Reply:
I don't, not quite sure what happened here.

---
In reply to: [199965797](https://github.com/dotnet/machinelearning/pull/462#discussion_r199965797) [](ancestors = 199965797)